.github/workflows/publish.yml

```diff
@@ -1,61 +1,99 @@
 # https://github.com/marketplace/actions/publish-nuget
 name: publish to nuget
 on:
+  workflow_dispatch:
   push:
     paths:
       - 'source/**'
       - '.github/**'
     branches:
       - main # Default release branch
 
+env:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+  DOTNET_NOLOGO: true
+  NuGetDirectory: ${{ github.workspace}}/nuget
+
+defaults:
+  run:
+    shell: pwsh
+
 jobs:
-  publish:
-    name: build, pack & publish
+  create_nuget:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - name: Setup .NET Core
-        uses: actions/setup-dotnet@v4
-        with:
-          dotnet-version: 8.0.201
-      # Publish
-      - name: publish tool on version change
-        id: publish_tool
-        uses: rohith/publish-nuget@v2
-        with:
-          # Filepath of the project to be packaged, relative to root of repository
-          PROJECT_FILE_PATH: source/SemVerAnalyzer/SemVerAnalyzer.csproj
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # Get all history to allow automatic versioning using MinVer
 
-          # NuGet package id, used for version detection & defaults to project name
-          # PACKAGE_NAME: Core
+    # Install the .NET SDK indicated in the global.json file
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
```

Comment:
Don't we need to specify `dotnet-version`?

Reply:
Looks like if we have a global.json with the SDK in it we don't need to specify it in the yml files